analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -211,22 +227,56 @@ any(duplicated(rna_df$sample_id))
 ```
 What are they?
 
+Checking for duplicate rna sample_ids
+
 ```{r}
 histologies_df %>%
   filter(sample_id %in% rna_df$sample_id[which(duplicated(rna_df$sample_id))],
          experimental_strategy == "RNA-Seq") %>%
   group_by(sample_id) %>%
-  summarize(RNA_library = paste(RNA_library, collapse = ", "))
+  summarize(RNA_library = paste(RNA_library, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
 ```
 
-There are 2 instances where there are two stranded biospecimen IDs that map to the same `sample_id`.
+The duplicates are mostly from multiple Solid tumor samples per sample_id or there are non-tumor ( derived cell-line ) samples for sample_id 
+
+
+Checking if adding cell-line composition type can uniquely match DNA to RNA
+
+```{r}
+
+cell_line_composition <- read_tsv("input/cell-line-composition.tsv",col_types = 
+                                    readr::cols( aliquot_id  = readr::col_character()))
+
+cell_line_composition %>%
+  left_join(histologies_df) %>%
+  group_by(sample_id,experimental_strategy) %>%
+  summarize(cell_line_composition = paste(cell_line_composition, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
+
+```
+`cell_line_composition` seems to be able to differentiate each cell line so let's add the column to merged dna and rna files
```

Comment:
```suggestion
`cell_line_composition` seems to be able to differentiate each cell line so let's add the column to merged dna and rna files
```

Agree with @jaclyn-taroni - can you show that sample ID and aliquot ID is not sufficient here?

Reply:
@kgaonkar6 I was going to request myself for another review, but I don't think the comments in this thread are fully addressed in the notebook itself yet.

Reply:
I think the DNA and RNA samples will always have different aliquot_id, it seems aliquot_id is associated with different experiemtal_strategy and composition (Cell Line, DNA, RNA, etc) , should I just add the aliquot_id column like below and comment that in the notebook?

DNA
sample_id<chr> | composition<chr> | aliquot_id<chr> 
-- | -- | -- 
7316-1746 | Derived Cell Line, Derived Cell Line | 739839, 739838 
7316-1763 | Derived Cell Line, Derived Cell Line, Solid Tissue | 739837, 739836, 549707
7316-1769 | Solid Tissue, Derived Cell Line | 549713, 739826


RNA

sample_id<chr> | RNA_library<chr> | composition<chr> | aliquot_id
-- | -- | -- | --
7316-1746 | stranded, stranded, stranded | Derived Cell Line, Derived Cell Line, Solid Tissue | 739942, 739941, 574552 |  
7316-1763 | stranded, stranded, stranded | Derived Cell Line, Derived Cell Line, Solid Tissue | 739939, 739940, 549574 |  
7316-1769 | stranded, stranded | Derived Cell Line, Solid Tissue | 740009, 470030 |  

Reply:
Got it - yes, if you can also add `composition_type` into the above tables and explain that this is required for matching because without that extra information, composition, sample_id, and aliquot_id cannot distinguish which cell line RNA/DNAs should be subtyped together.

Reply:
added this bit to the code now